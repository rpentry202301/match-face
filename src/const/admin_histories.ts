export const skills = [
    {id:1,name:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,name:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,name:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,name:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,name:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,name:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,name:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,name:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,name:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,name:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const answer_request_groups =[
    {id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]
