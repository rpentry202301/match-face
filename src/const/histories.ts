export type DataType = {
  id: number;
  comment_created_at: boolean;
  name: string;
  detail: string;
  answer_update_at: string;
  skill_id: number[];
}[];

export type SkillType = {
  id: number;
  name: string;
  createdUser: string;
  createdAt: string;
  updateUser: string;
  updateAt: string;
}[];

export const projects: DataType = [
  {
    id: 1,
    comment_created_at: true,
    name: "ポムポムプリン",
    detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
  {
    id: 2,
    comment_created_at: false,
    name: "日暮らし、",
    detail:
      "硯にむかひて、心にうつりゆくよしなし事を、そこはかとなく書きつくれば、あやしうこそものぐるほしけれ。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [2, 3, 4],
  },
  {
    id: 3,
    comment_created_at: false,
    name: "親譲りの無鉄砲で",
    detail:
      "小供の時から損ばかりしている。小学校に居る時分学校の二階から飛び降りて一週間ほど腰を抜かした事がある。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [5, 6, 7],
  },
  {
    id: 4,
    comment_created_at: true,
    name: "なぜそんな無闇を",
    detail:
      "したと聞く人があるかも知れぬ。別段深い理由でもない。新築の二階から首を出していたら、同級生の一人が冗談に、いくら威張っても、そこから飛び降りる事は出来まい。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [1, 2, 6, 7],
  },
  {
    id: 5,
    comment_created_at: true,
    name: "ポムポムプリン",
    detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [3, 4, 5],
  },
  {
    id: 6,
    comment_created_at: false,
    name: "弱虫やーい。",
    detail:
      "と囃したからである。小使に負ぶさって帰って来た時、おやじが大きな眼をして二階ぐらいから飛び降りて腰を抜かす奴があるかと云ったから",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
  {
    id: 7,
    comment_created_at: true,
    name: "つれづれなるまゝ",
    detail:
      "日暮らし、硯にむかひて、心にうつりゆくよしなし事を、そこはかとなく書きつくれば、あやしうこそものぐるほしけれ。",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
];

// export {data};

export const skills = [
  {
    id: 1,
    skill: "JavaScript",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 2,
    skill: "TypeScript",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 3,
    skill: "React",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 4,
    skill: "Next.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 5,
    skill: "Vue.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 6,
    skill: "Angular",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 7,
    skill: "Nuxt.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 8,
    skill: "jQuery",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 9,
    skill: "Redux",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 10,
    skill: "vuex",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 11,
    skill: "MaterialUI",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 12,
    skill: "tailwind",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 13,
    skill: "Bootstrap",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 14,
    skill: "Vercel",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
];
