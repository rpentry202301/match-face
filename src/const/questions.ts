type Data = {
  id: string;
  project_name: string;
  project_detail: string;
  answer_deadline: string;
  answer_status: boolean;
}[];

export const mockData = [
  {
    id: 1,
    deadline: "2023-12-01 18:00",
    project: {
      name: "バックエンド案件",
      detail: "販促アプリの新規開発、既存システムの保守・運用。",
    },
    answered: false,
  },
];

// export const projects = [
//   {
//     id: 1,
//     name: "案件A",
//     detail: "案件Aの説明",
//     enterprise_id: 1,
//     department_id: 1,
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 2,
//     name: "案件B",
//     detail: "案件Bの説明",
//     enterprise_id: 1,
//     department_id: 1,
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 3,
//     name: "案件C",
//     detail: "案件Cの説明",
//     enterprise_id: 1,
//     department_id: 2,
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 4,
//     name: "案件D",
//     detail: "案件Dの説明",
//     enterprise_id: 1,
//     department_id: 1,
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 5,
//     name: "案件E",
//     detail: "案件Eの説明",
//     enterprise_id: 1,
//     department_id: 2,
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
// ];

// export const answer_requests = [
//   {
//     id: 1,
//     user_id: [1, 2],
//     administrator_id: 5,
//     project_id: 1,
//     request_at: "2023-07-10T16:52:46.053Z",
//     deadline: "2023-07-10T16:52:46.053Z",
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 2,
//     user_id: [1, 2],
//     administrator_id: 5,
//     project_id: 2,
//     request_at: "2023-07-10T16:52:46.053Z",
//     deadline: "2023-07-10T16:52:46.053Z",
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 3,
//     user_id: [3],
//     administrator_id: 5,
//     project_id: 3,
//     request_at: "2023-07-10T16:52:46.053Z",
//     deadline: "2023-07-15T16:52:46.053Z",
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 4,
//     user_id: [1, 2, 4],
//     administrator_id: 5,
//     project_id: 4,
//     request_at: "2023-07-10T16:52:46.053Z",
//     deadline: "2023-07-15T16:52:46.053Z",
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 5,
//     user_id: [4],
//     administrator_id: 5,
//     project_id: 5,
//     request_at: "2023-07-10T16:52:46.053Z",
//     deadline: "2023-08-10T16:52:46.053Z",
//     created_user: "rakus",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "rakus",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
// ];

// export const answer_requests_questions = [
//   // 1~5はrequest_idは１。全てのquestionsが回答済みの想定。
//   {
//     id: 1,
//     question_id: 1,
//     answer_request_id: 1,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 2,
//     question_id: 2,
//     answer_request_id: 1,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 3,
//     question_id: 3,
//     answer_request_id: 1,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 4,
//     question_id: 4,
//     answer_request_id: 1,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 5,
//     question_id: 5,
//     answer_request_id: 1,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   // 6~10はquestion_id(回答依頼ID)が２。１つ以上未回答を含む。
//   {
//     id: 6,
//     question_id: 1,
//     answer_request_id: 2,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 7,
//     question_id: 2,
//     answer_request_id: 2,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 8,
//     question_id: 3,
//     answer_request_id: 2,
//     is_answered: true,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 9,
//     question_id: 4,
//     answer_request_id: 2,
//     is_answered: false,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
//   {
//     id: 10,
//     question_id: 5,
//     answer_request_id: 2,
//     is_answered: false,
//     created_user: "ラクス太郎",
//     created_at: "2023-07-10T16:52:46.053Z",
//     update_user: "ラクス太郎",
//     update_at: "2023-07-10T16:52:46.053Z",
//   },
// ];
