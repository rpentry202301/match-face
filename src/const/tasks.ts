import { Answer_RequestsType } from "@/types/admin/tasks/types";

// テスト用データ
export const departments = [
  {
    id: 1,
    name: "Java",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 2,
    name: "PHP",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 3,
    name: "FR",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 4,
    name: "CL",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 5,
    name: "ML",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 6,
    name: "QA",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
];

export const answer_requests: Answer_RequestsType = {
  answerRequests: [
    {
      id: 1,
      project: {
        id: 1,
        name: "バックエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 1,
        name: "Java",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 1,
          userId: 1,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
    {
      id: 2,
      project: {
        id: 2,
        name: "バックエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 2,
        name: "PHP",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 2,
          userId: 2,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
    {
      id: 3,
      project: {
        id: 3,
        name: "フロントエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 3,
        name: "FR",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 3,
          userId: 3,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
  ],
};
