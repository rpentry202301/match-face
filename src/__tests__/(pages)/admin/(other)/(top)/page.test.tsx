import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import AdminHome from "@/app/(pages)/admin/(other)/(top)/page";
import React from "react";
import "@testing-library/jest-dom";
import { dataArray } from "@/const/adminTop";

describe("adminトップページのテスト", () => {
  const user = userEvent.setup();
  beforeEach(() => {
    render(<AdminHome />);
  });
  it("スナップショット", async () => {
    const { container } = render(<AdminHome />);
    expect(container).toMatchSnapshot();
  });

  it("データとメニューの数が等しい", async () => {
    const menuComponents = screen.getAllByTestId("menu_link");
    // console.log(menuComponents.length);
    expect(menuComponents.length).toBe(dataArray.length);
  });
  it("メニューが正しく表示される", () => {
    const title = screen.getByText(dataArray[0].title);
    const url = document.querySelector(`a[href="${dataArray[0].url}"]`);
    const description = screen.getByText(dataArray[0].description);
    const img = document.querySelector("img");
    // console.log(img);
    const imgAlt = img?.getAttribute("alt");
    // console.log(imgAlt);
    const imgUrl = img?.getAttribute("src");
    // console.log(imgUrl);

    expect(title).toBeInTheDocument();
    expect(url).toBeInTheDocument();
    expect(description).toBeInTheDocument();
    expect(imgAlt).toContain(dataArray[0].imgAlt);
    expect(imgUrl).toContain("document_icon.png");
  });

  it("メニュークリックでページ遷移", () => {
    const menu = document.querySelector(
      `a[href="${dataArray[0].url}"]`
    ) as HTMLElement;
    // console.log(menu);
    user.click(menu);
    expect(menu).toHaveAttribute("href", "/admin/handle-question");
  });
});
