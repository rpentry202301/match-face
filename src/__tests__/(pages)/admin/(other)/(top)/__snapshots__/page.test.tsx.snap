// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adminトップページのテスト スナップショット 1`] = `
<div>
  <div
    class="flex justify-center items-center min-h-screen md:mt-[-40px]"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-3 md:mt-10"
    >
      <div
        class="m-10"
      >
        <a
          href="/admin/handle-question"
        >
          <button
            class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
            data-testid="menu_link"
          >
            <p
              class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
            >
              質問・回答の追加・編集
            </p>
            <div
              class="flex pt-4 w-full"
            >
              <p
                class="text-left mr-auto"
              >
                質問・回答例を追加・編集します。
              </p>
              <img
                alt="document_icon"
                class="pt-2 pl-2"
                data-nimg="1"
                decoding="async"
                height="70"
                loading="lazy"
                src="/_next/image?url=%2Ficon%2Fdocument_icon.png&w=256&q=75"
                srcset="/_next/image?url=%2Ficon%2Fdocument_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fdocument_icon.png&w=256&q=75 2x"
                style="color: transparent;"
                width="70"
              />
            </div>
          </button>
        </a>
      </div>
      <div
        class="m-10"
      >
        <a
          href="/admin/tasks"
        >
          <button
            class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
            data-testid="menu_link"
          >
            <p
              class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
            >
              タスクの一覧
            </p>
            <div
              class="flex pt-4 w-full"
            >
              <p
                class="text-left mr-auto"
              >
                タスクの一覧を確認できます。
              </p>
              <img
                alt="pencil_icon"
                class="pt-2 pl-2"
                data-nimg="1"
                decoding="async"
                height="70"
                loading="lazy"
                src="/_next/image?url=%2Ficon%2Fpencil_icon.png&w=256&q=75"
                srcset="/_next/image?url=%2Ficon%2Fpencil_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fpencil_icon.png&w=256&q=75 2x"
                style="color: transparent;"
                width="70"
              />
            </div>
          </button>
        </a>
      </div>
      <div
        class="m-10"
      >
        <a
          href="/admin/histories"
        >
          <button
            class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
            data-testid="menu_link"
          >
            <p
              class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
            >
              回答の閲覧
            </p>
            <div
              class="flex pt-4 w-full"
            >
              <p
                class="text-left mr-auto"
              >
                ユーザーの回答を閲覧できます。
              </p>
              <img
                alt="search_icon"
                class="pt-2 pl-2"
                data-nimg="1"
                decoding="async"
                height="70"
                loading="lazy"
                src="/_next/image?url=%2Ficon%2Fsearch_icon.png&w=256&q=75"
                srcset="/_next/image?url=%2Ficon%2Fsearch_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fsearch_icon.png&w=256&q=75 2x"
                style="color: transparent;"
                width="70"
              />
            </div>
          </button>
        </a>
      </div>
      <div
        class="m-10"
      >
        <a
          href="/admin/users"
        >
          <button
            class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
            data-testid="menu_link"
          >
            <p
              class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
            >
              ユーザー管理
            </p>
            <div
              class="flex pt-4 w-full"
            >
              <p
                class="text-left mr-auto"
              >
                ユーザーの一覧を確認できます。
              </p>
              <img
                alt="user_icon"
                class="pt-2 pl-2"
                data-nimg="1"
                decoding="async"
                height="70"
                loading="lazy"
                src="/_next/image?url=%2Ficon%2Fuser_icon.png&w=256&q=75"
                srcset="/_next/image?url=%2Ficon%2Fuser_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fuser_icon.png&w=256&q=75 2x"
                style="color: transparent;"
                width="70"
              />
            </div>
          </button>
        </a>
      </div>
      <div
        class="m-10"
      >
        <a
          href="/admin/groups"
        >
          <button
            class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
            data-testid="menu_link"
          >
            <p
              class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
            >
              グループ一覧
            </p>
            <div
              class="flex pt-4 w-full"
            >
              <p
                class="text-left mr-auto"
              >
                グループの一覧を確認できます。
              </p>
              <img
                alt="group_icon"
                class="pt-2 pl-2"
                data-nimg="1"
                decoding="async"
                height="70"
                loading="lazy"
                src="/_next/image?url=%2Ficon%2Fgroup_icon.png&w=256&q=75"
                srcset="/_next/image?url=%2Ficon%2Fgroup_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fgroup_icon.png&w=256&q=75 2x"
                style="color: transparent;"
                width="70"
              />
            </div>
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
